app/user/serializers.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes, force_str
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.validators import UniqueValidator
+
+from app.user.models import Profile
+from app.user.token import account_activation_token
+from app.user.validators import (
+    validate_password_digit,
+    validate_password_lowercase,
+    validate_password_symbol,
+    validate_password_uppercase,
+)
+from speaksfer.settings.base import EMAIL_USER
+
+User = get_user_model()
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    username = serializers.CharField(
+        max_length=20,
+        min_length=8,
+        validators=[UniqueValidator(queryset=User.objects.all())],
+    )
+
+    email = serializers.EmailField(
+        required=True,
+        validators=[UniqueValidator(queryset=User.objects.all())],
+    )
+
+    password = serializers.CharField(
+        max_length=128,
+        min_length=8,
+        write_only=True,
+        validators=[
+            validate_password_digit,
+            validate_password_uppercase,
+            validate_password_symbol,
+            validate_password_lowercase,
+        ],
+    )
+
+    class Meta:
+        model = User
+        fields = ("email", "username", "password")
+
+    @staticmethod
+    def send_email(user: Any, request: Request) -> None:
+
+        current_site_info = get_current_site(request)
+        email_body = render_to_string(
+            "email_verification.html",
+            {
+                "user": user,
+                "domain": current_site_info.domain,
+                "uid": urlsafe_base64_encode(force_bytes(user.pk)),
+                "token": account_activation_token.make_token(user),
+            },
+        )
+
+        send_mail(
+            "Verify  your email!",
+            email_body,
+            EMAIL_USER,
+            [user.email],
+            fail_silently=False,
+        )
+
+    def create(self, validated_data: Any) -> Any:
+        request = self.context.get("request")
+        user = User.objects.create_user(**validated_data)
+        self.send_email(user, request)  # type: ignore
```

Comment:
No need for ignoring mypy here, fix the error

Reply:
This was one of the ones I tried to resolve but it kept throwing an error